Manito/Manito/Screens/CreateRoom/View/CreateRoomView.swift

```diff
@@ -0,0 +1,348 @@
+        self.nextButton.snp.makeConstraints {
+            $0.leading.trailing.equalTo(self.safeAreaLayoutGuide).inset(Size.leadingTrailingPadding)
+            $0.bottom.equalTo(self.safeAreaLayoutGuide).inset(23)
+            $0.height.equalTo(60)
+        }
+        
+        self.addSubview(self.roomTitleView)
+        self.addSubview(self.roomCapacityView)
+        self.addSubview(self.roomDateView)
+        self.addSubview(self.roomInfoView)
+        self.addSubview(self.characterCollectionView)
+        
+        [self.roomTitleView, self.roomCapacityView, self.roomDateView, self.roomInfoView, self.characterCollectionView]
+            .forEach {
+                $0.snp.makeConstraints {
+                    $0.top.equalTo(self.titleLabel.snp.bottom).offset(66)
+                    $0.leading.trailing.equalToSuperview().inset(Size.leadingTrailingPadding)
+                    $0.bottom.equalTo(self.nextButton.snp.top)
+                }
+            }
+        
+        self.bringSubviewToFront(self.nextButton)
+    }
+    
+    private func setupAction() {
+        let closeAction = UIAction { [weak self] _ in
+            self?.delegate?.didTapCloseButton()
+        }
+        self.closeButton.addAction(closeAction, for: .touchUpInside)
+        
+        let nextAction = UIAction { [weak self] _ in
+            guard let currentStep = self?.roomStep else { return }
+            guard let nextStep = self?.moveToNextStep(currentStep: currentStep) else { return }
+            
+            self?.runActionAtStep(at: currentStep)
+            self?.roomStep = nextStep
+        }
+        self.nextButton.addAction(nextAction, for: .touchUpInside)
+        
+        let backAction = UIAction { [weak self] _ in
+            guard let currentStep = self?.roomStep else { return }
+            guard let previousStep = self?.moveToPreviousStep(currentStep: currentStep) else { return }
+            self?.roomStep = previousStep
+        }
+        self.backButton.addAction(backAction, for: .touchUpInside)
+    }
+    
+    private func setupNotificationCenter() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillShow),
+                                               name: UIResponder.keyboardWillShowNotification,
+                                               object: nil)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillHide),
+                                               name: UIResponder.keyboardWillHideNotification,
+                                               object: nil)
+    }
+    
+    private func moveToNextStep(currentStep: CreateRoomStep) -> CreateRoomStep {
+        return currentStep.nextStep()
+    }
+    
+    private func moveToPreviousStep(currentStep: CreateRoomStep) -> CreateRoomStep {
+        return currentStep.previousStep()
+    }
```

Comment:
p4;

저도 지난번에 작성할 때 미처 생각치 못했던 부분인데

<img width="828" alt="image" src="https://github.com/DeveloperAcademy-POSTECH/MC2-Team5-Firefighter/assets/78677571/98324f5d-d8f4-4f64-91aa-97c2794a506f">

출처 : https://yoonah-dev.oopy.io/ea3ba951-ca0b-4b8b-b6e3-95fdae0e135b

요거를 보고 나니까 더 다듬을 수 있는 부분인 것 같아서요!

```suggestion
    private func moveToNextStep(from step: CreateRoomStep) -> CreateRoomStep {
        return step.next()
    }
    
    private func moveToPreviousStep(from step: CreateRoomStep) -> CreateRoomStep {
        return step.previous()
    }
```

그리고 아래에 extension부분에서는

```swift
private extension CreateRoomView.CreateRoomStep {
    func next() -> Self {
        switch self {
        // 생략
        }
    }

    func previous() -> Self {
        switch self {
        // 생략
        }
    }
}
```

netxStep()을 next()로 previousStep()을 previous()로 수정하면 어떨까요??

투머치라고 생각이 되거나 다르게 생각하신다면 말씀해주세용 ^_________^

Reply:
좋습니다 이해가 팍 됐습니다. 

투머치 하지 않고 오히려 디테일이라고 생각이 들어 감사하네유